libs/common/result.hpp

```diff
@@ -81,21 +81,38 @@ namespace iroha {
        * @nocode
        */
       template <typename ValueMatch, typename ErrorMatch>
-      constexpr auto match(ValueMatch &&value_func, ErrorMatch &&error_func) {
+      constexpr auto match(ValueMatch &&value_func, ErrorMatch &&error_func) & {
```

Comment:
I guess this and && and const && function modifiers should be documented.

Reply:
do you mean copy-pasting the docstrings? maybe we can better make some sort of reference?